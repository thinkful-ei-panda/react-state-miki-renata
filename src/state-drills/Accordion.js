import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {
        sections: []
    };

    constructor(props) {
        super(props)
        this.state = {
            currentSection: ""
        };
    };

    handleButtonClick(i) {
        this.setState({
            currentSection: this.props.sections[i].content
        });
    };


    renderContent(i) {
        return (
            <p key={i}>
                {this.state.currentSection}
            </p>
        );
    };

    renderButtons() {
        return this.props.sections.map((section, i) => (
            <li>
                 <button key={i} onClick={() => this.handleButtonClick(i)}>
                {section.title}
                </button>
                {this.state.currentSection === section.content? this.renderContent(i) : null}
            </li>
        ));
    };

    render() {
      return (
            <ul>
                {this.renderButtons()}
            </ul>
        );
    };
};

export default Accordion;