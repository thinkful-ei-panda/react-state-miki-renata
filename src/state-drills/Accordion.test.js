import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

describe('Accordion component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion sections={sections} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<Accordion sections={sections} />)
          .toJSON();
        expect(tree).toMatchSnapshot();  
    });

    // Create wrapper instance
    // Transform wrapper into JSON object
    // Compare with snapshot

    it('renders empty given no sections', () => {
        // The wrapper allows us to find nodes within the component and to simulate events
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    
    it('renders no sections by default', () => {
        const wrapper = shallow(<Accordion sections={sections} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    // Test for clicking the button
    // Create wrapper instance
    // Traverse to button
    
    it('closes the first tab and opens any clicked tab', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        // .debug() lets you see the HTML for selected node
        console.log('>>> WRAPPER <<<');
        console.log(wrapper.debug());
        console.log('>>> FIND(BUTTON) <<<');
        console.log(wrapper.find('button').at(1).debug());
        // .find() to traverse to a node specified by selector
        // .at(index) lets you select a specific selector (of many similar ones)
        // .simulate('event') to simulate some event
        wrapper.find('button').at(1).simulate('click');
        // Expected output as a result of event
        expect(toJson(wrapper)).toMatchSnapshot();
    });





})