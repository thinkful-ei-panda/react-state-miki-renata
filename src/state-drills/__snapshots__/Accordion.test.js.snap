// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component closes the first tab and opens any clicked tab 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <p
      key="1"
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders empty given no sections 1`] = `<ul />`;

exports[`Accordion component renders no sections by default 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders the UI as expected 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
